test/cpp/jit/test_misc.cpp

```diff
@@ -2048,6 +2048,52 @@ void testFutures() {
     s2->markCompleted(1);
     ASSERT_EQ(c4->value().toInt(), 7);
   }
+
+  bool cbFinished = false;
```

Comment:
maybe it's worth separating these TLSState-oriented tests in a separate test function?
(add the test function in caffe2/test/cpp/jit/tests.h)